pynxtools/dataconverter/writer.py

```diff
@@ -279,6 +279,9 @@ def add_units_key(dataset, path):
                               f"with the following message: {str(exc)}") from exc
         for links in hdf5_links_for_later:
             dataset = handle_dicts_entries(*links)
+            if dataset is None:
+                # If target of a link is invalid to be linked
+                del self.data[links[-1]]
```

Comment:
Shouldn't this delete the current path we are checking i.e. the path that holds the links dict in the template not the target? What is links[-1] return?